ipatests/test_integration/test_installation.py

```diff
@@ -1119,6 +1130,115 @@ def run_request(url, expected_stdout=None, expected_stderr=None):
+                               'DNS (BIND)? [no]: ')
+                e.sendline('no')
+                e.expect_exact('Please confirm the domain name [{}]: '.format(
+                    self.master.hostname
+                ))
+                e.sendline(self.master.domain.name)
+                e.expect_exact('Please provide a realm name [{}]: '.format(
+                    self.master.domain.realm
+                ))
+                e.sendline(self.master.domain.realm)
+                e.expect_exact('Directory Manager password: ')
+                e.sendline(self.master.config.dirman_password)
+                e.expect_exact('Password (confirm): ')
+                e.sendline(self.master.config.dirman_password)
+                e.expect_exact('IPA admin password: ')
+                e.sendline(self.master.config.admin_password)
+                e.expect_exact('Password (confirm): ')
+                e.sendline(self.master.config.admin_password)
+                e.expect_exact('Do you want to configure chrony with '
+                               'NTP server or pool address? [no]: ')
+                e.sendline('no')
+                e.expect_exact('Continue to configure the system '
+                               'with these values? [no]: ')
+                e.sendline('yes')
+                e.expect_exit(ignore_remaining_output=True, timeout=720)
+                hostname = self.master.run_command([
+                    'hostname', '-f']).stdout_text.strip()
+            assert hostname == new_hostname
+        finally:
+            self.master.hostname = original_hostname
+            self.master.put_file_contents('/etc/hosts', hosts)
+
+    def test_ad_subpackage_dependency(self, server_cleanup):
+        """
+        Test if the installer is not dependant on trust-ad package and
+        succeeds even when AD subpackage is not installed
+
+        https://pagure.io/freeipa/issue/4011
+        """
+        server_cleanup(self.master)
+        package_name = "*ipa-server-trust-ad"
+        tasks.uninstall_packages(self.master, [package_name])
+        tasks.install_master(self.master)
+
+    def test_backup_of_cs_cfg_should_be_created(self, server_cleanup):
+        """
+        Test that the installer backs up CS.cfg configuration before it's
+        being modified.
+
+        https://pagure.io/freeipa/issue/4166
+        """
+        server_cleanup(self.master)
+        bcp_location = paths.CA_CS_CFG_PATH + ".ipabkp"
+        tasks.install_master(self.master)
+        ipaserver_install_log = str(self.master.get_file_contents(
+            paths.IPASERVER_INSTALL_LOG, encoding='utf-8'))
+        assert "backing up CS.cfg" in ipaserver_install_log
+        try:
+            self.master.get_file_contents(bcp_location, encoding='utf-8')
+        except IOError:
+            pytest.fail("Backup of CS.cfg not created")
+            raise
+
+    @pytest.mark.skip_if_container(
```

Comment:
It seems to me that this test case can be moved to `TestInstallMaster` class which installs ipa server in the `install()` method. 
By removing `server_cleanup` and `tasks.install_master(self.master)` we can save about 10 minutes without degrading the test.

Reply:
I am sorry, but I have to disagree. The tests must not be tight-coupled, even if it would mean extra machine time. Also, I am not sure what you meant, because `TestInstallMaster` has only `pass` in its class install method.